File: Récupération 20210215/Medial Supervisor 1.4.4.6/Module/mFileFunction.vb
Option Strict On
Option Explicit On

Module mFileFunction

    Public Const PATH As Short = 4
    Public Const FILENAME As Short = 3
    Public Const EXTENSION_ONLY As Short = 2
    Public Const FILENAME_ONLY As Short = 1
    Public Const PATH_FILENAME_ONLY As Short = 8

    Public Function ParsePath(ByVal szPath As String, ByVal nOperation As Short) As String

        Dim nPosSlash, nPosPoint, nLenPath As Integer

        nPosPoint = InStrRev(szPath, ".") - 1
        nPosSlash = InStrRev(szPath, "\")
        nLenPath = Len(szPath)

        If nOperation = EXTENSION_ONLY Then
            Return Right(szPath, nLenPath - nPosPoint)
        ElseIf nOperation = FILENAME_ONLY Then
            Return Mid(szPath, nPosSlash + 1, nPosPoint - nPosSlash)
        ElseIf nOperation = FILENAME Then
            Return Right(szPath, nLenPath - nPosSlash)
        ElseIf nOperation = PATH Then
            Return Left(szPath, nPosSlash)
        ElseIf nOperation = PATH_FILENAME_ONLY Then
            Return Left(szPath, nPosPoint)
        Else
            Return szPath
        End If


    End Function

    Public Function StrAddLastSlash(ByVal svString As String) As String

        If (svString <> "") Then
            If Right(svString, 1) <> "\" Then svString &= "\"
        End If

        Return svString

    End Function

End Module

File: JFCUpdateService/JFCUpdateService/mFileFunction.cs
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace JFCUpdateService
{
    [StandardModule]
    internal sealed class mFileFunction
    {
        public const short PATH = 4;

        public const short FILENAME = 3;

        public const short EXTENSION_ONLY = 2;

        public const short FILENAME_ONLY = 1;

        public const short PATH_FILENAME_ONLY = 8;

        public static string ParsePath(string szPath, short nOperation)
        {
            checked
            {
                int num = Strings.InStrRev(szPath, ".") - 1;
                int num2 = Strings.InStrRev(szPath, "\\");
                int num3 = Strings.Len(szPath);
                switch (nOperation)
                {
                    case 2:
                        return Strings.Right(szPath, num3 - num);
                    case 1:
                        return Strings.Mid(szPath, num2 + 1, num - num2);
                    case 3:
                        return Strings.Right(szPath, num3 - num2);
                    case 4:
                        return Strings.Left(szPath, num2);
                    case 8:
                        return Strings.Left(szPath, num);
                    default:
                        return szPath;
                }
            }
        }

        public static string StrAddLastSlash(string svString)
        {
            if (Operators.CompareString(svString, "", TextCompare: false) != 0 && Operators.CompareString(Strings.Right(svString, 1), "\\", TextCompare: false) != 0)
            {
                svString += "\\";
            }
            return svString;
        }
    }
}